### ISSUE TITLE
Misleading error message on invalid route: "default milestones requires the directions route to be requested with voice instructions enabled"

### ISSUE THREAD

**Issue by Reiszecke** [mapbox/mapbox-navigation-android#2034]
Old title: "Crash due to "default milestones requires the directions route to be requested with voice instructions enabled". How to set those?"

Trying to work through the examples I'm now trying to display a route returned by the API:

```MapboxDirections client = MapboxDirections.builder()
            .origin(start)
            .destination(destination)
            .overview(DirectionsCriteria.OVERVIEW_FULL)
            .profile(DirectionsCriteria.PROFILE_CYCLING)
            .accessToken("000")
            .build();

        client.enqueueCall(new Callback<DirectionsResponse>() {

...

                NavigationLauncherOptions options = NavigationLauncherOptions.builder()
                    .directionsRoute(currentRoute)
                    .shouldSimulateRoute(false)
                    .build();

                NavigationLauncher.startNavigation(MainActivity.this, options);
```

This results in a crash.

```2019-08-27 03:46:27.120 2555-2555/? E/Mbgl-MapChangeReceiver: Exception in onDidFinishLoadingStyle
    java.util.MissingFormatArgumentException: Format specifier 'Using the default milestones requires the directions route to be requested with voice instructions enabled.'
        at com.mapbox.services.android.navigation.v5.utils.ValidationUtils.checkInvalidVoiceInstructions(ValidationUtils.java:36)
        at com.mapbox.services.android.navigation.v5.utils.ValidationUtils.validDirectionsRoute(ValidationUtils.java:19)
        at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation.startNavigationWith(MapboxNavigation.java:938)
        at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation.startNavigation(MapboxNavigation.java:346)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel.startNavigation(NavigationViewModel.java:434)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel.updateRoute(NavigationViewModel.java:251)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewRouteEngineListener.onRouteUpdate(NavigationViewRouteEngineListener.java:17)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewRouter.updateCurrentRoute(NavigationViewRouter.java:96)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewRouter.extractRouteFrom(NavigationViewRouter.java:121)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewRouter.extractRouteOptions(NavigationViewRouter.java:72)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel.initialize(NavigationViewModel.java:213)
        at com.mapbox.services.android.navigation.ui.v5.NavigationView.initializeNavigation(NavigationView.java:618)
        at com.mapbox.services.android.navigation.ui.v5.NavigationView.startNavigation(NavigationView.java:381)
        at com.mapbox.services.android.navigation.ui.v5.MapboxNavigationActivity.onNavigationReady(MapboxNavigationActivity.java:99)
        at com.mapbox.services.android.navigation.ui.v5.NavigationView$1.onStyleLoaded(NavigationView.java:225)
        at com.mapbox.mapboxsdk.maps.MapboxMap.notifyStyleLoaded(MapboxMap.java:855)
        at com.mapbox.mapboxsdk.maps.MapboxMap.onFinishLoadingStyle(MapboxMap.java:212)
        at com.mapbox.mapboxsdk.maps.MapView$MapCallback.onDidFinishLoadingStyle(MapView.java:1264)
        at com.mapbox.mapboxsdk.maps.MapChangeReceiver.onDidFinishLoadingStyle(MapChangeReceiver.java:198)
        at com.mapbox.mapboxsdk.maps.NativeMapView.onDidFinishLoadingStyle(NativeMapView.java:1035)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:323)
        at android.os.Looper.loop(Looper.java:136)
        at android.app.ActivityThread.main(ActivityThread.java:6165)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:888)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:778)
    
    
    --------- beginning of crash
2019-08-27 03:46:27.120 2555-2555/? A/libc: /usr/local/google/buildbot/src/android/ndk-release-r19/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:73: abort_message: assertion "terminating with uncaught exception of type jni::PendingJavaException" failed
2019-08-27 03:46:27.121 2555-2555/? A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 2555 (s.driverapptest)
```

I thought ```        RouteOptions routeOptions = RouteOptions.builder()
            .voiceInstructions(true)
            .build();``` might fix this, however I have no idea where do pass this. The iOS examples got this covered, not so much with Android. How can I get around this issue?

**Comment by Reiszecke** [comment]
Fixed it, Java switches x and y hence the found route (probably somewhere in the ocean) had a length of 0. That's when it hit me. The error message is definitely misleading tho.

**Comment by BenF99** [comment]
Encountering the same error - any chance you could explain your fix? Thanks

**Comment by lassons** [comment]
Identical problem, we have bloqued ...

**Comment by lassons** [comment]
Not totaly resolved, also, i have another error

You can test to add                

 .bannerInstructions(true)
 .voiceInstructions(true)

inside MapboxDirections.builder()
example : 
        MapboxDirections client = MapboxDirections.builder()
                .origin(directionsOriginPoint)
                .destination(destination)
                .overview(DirectionsCriteria.OVERVIEW_FULL)
                .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
                .bannerInstructions(true)
                .voiceInstructions(true)
                .accessToken(getString(R.string.access_token))
                .language(Locale.FRANCE)
                .build();


best regards

Sébastien LASSON

**Comment by Guardiola31337** [comment]
Hey @Reiszecke @BenF99 @lassons 👋 thanks for checking out the SDK.

Closing here as milestones are not going to be available in `1.0`. Stay tuned for the public beta release 🚀 

This also references https://github.com/mapbox/mapbox-navigation-android/issues/3034

**Comment by Guardiola31337** [comment]
Hey @Reiszecke @BenF99 @lassons we've just released [`release_core_1.0.0-rc.1`](https://github.com/mapbox/mapbox-navigation-android/releases/tag/release_core_1.0.0-rc.1) 🎉 🚀 

Please review the [developer documentation](https://docs.mapbox.com/android/beta/navigation/overview/#installation) to start building with the Mapbox core Navigation SDK v1.0 for Android and if you already use an older version of the Navigation SDK, check out [the migration guide](https://github.com/mapbox/mapbox-navigation-android/wiki/1.0-Navigation-SDK-Migration-Guide) to transition your project from the "legacy" core Navigation SDK to the 1.0 version and let us know if you have any questions or run into issues and please open tickets in https://github.com/mapbox/mapbox-navigation-android/issues/new and we will take it from there! We would love your feedback.

Thanks again!
